homeassistant/components/compensation/__init__.py

```diff
@@ -87,59 +91,100 @@ def datapoints_greater_than_degree(value: dict) -> dict:
 )
 
 
+async def create_compensation_data(
+    hass: HomeAssistant, compensation: str, conf: ConfigType, should_raise: bool = False
+) -> None:
+    """Create compensation data."""
+    _LOGGER.debug("Setup %s.%s", DOMAIN, compensation)
+
+    degree = conf[CONF_DEGREE]
+
+    initial_coefficients: list[tuple[float, float]] = conf[CONF_DATAPOINTS]
+    sorted_coefficients = sorted(initial_coefficients, key=itemgetter(0))
+
+    # get x values and y values from the x,y point pairs
+    x_values, y_values = zip(*initial_coefficients, strict=False)
+
+    # try to get valid coefficients for a polynomial
+    coefficients = None
+    with np.errstate(all="raise"):
+        try:
+            coefficients = np.polyfit(x_values, y_values, degree)
+        except FloatingPointError as error:
+            _LOGGER.error(
+                "Setup of %s encountered an error, %s",
+                compensation,
+                error,
+            )
+            if should_raise:
+                raise ConfigEntryError(
+                    translation_domain=DOMAIN,
+                    translation_key="setup_error",
+                    translation_placeholders={
+                        "title": conf[CONF_NAME],
+                        "error": str(error),
+                    },
+                ) from error
```

Comment:
```suggestion
    with np.errstate(all="raise"):
    try:
        coefficients := np.polyfit(x_values, y_values, degree)
    except FloatingPointError as error:
        _LOGGER.error(
            "Setup of %s encountered an error, %s",
            compensation,
            error,
        )
        if should_raise:
            raise ConfigEntryError(
                translation_domain=DOMAIN,
                translation_key="setup_error",
                translation_placeholders={
                    "title": conf[CONF_NAME],
                    "error": str(error),
                },
            ) from error
```
Just preference I guess, but will a walrus operator you can skip the `coefficients = None`.